firestore/integration_test_internal/src/aggregate_query_snapshot_test.cc

```diff
@@ -0,0 +1,283 @@
+  return snapshot.Hash();
+}
+
+namespace {
+
+TEST_F(AggregateQuerySnapshotTest, DefaultConstructor) {
+  AggregateQuerySnapshot snapshot;
+  EXPECT_EQ(snapshot.query(), AggregateQuery());
+  EXPECT_EQ(snapshot.count(), 0);
+  EXPECT_FALSE(snapshot.is_valid());
+}
+
+TEST_F(AggregateQuerySnapshotTest, CopyConstructor) {
+  const Query& query = TestFirestore()->Collection("foo").Limit(10);
+  const AggregateQuery& aggregate_query = query.Count();
+
+  static const int COUNT = 5;
+  AggregateQuerySnapshot snapshot =
+      TestAggregateQuerySnapshot(aggregate_query, COUNT);
+
+  AggregateQuerySnapshot copied_snapshot(snapshot);
+
+  EXPECT_EQ(snapshot.count(), COUNT);
+  EXPECT_EQ(snapshot.query(), aggregate_query);
+  EXPECT_TRUE(snapshot.is_valid());
+
+  EXPECT_EQ(copied_snapshot.count(), COUNT);
+  EXPECT_EQ(copied_snapshot.query(), aggregate_query);
+  EXPECT_TRUE(copied_snapshot.is_valid());
+}
+
+TEST_F(AggregateQuerySnapshotTest, CopyAssignmentOperator) {
+  const Query& query = TestFirestore()->Collection("foo").Limit(10);
+  const AggregateQuery& aggregate_query = query.Count();
+
+  static const int COUNT = 7;
+  AggregateQuerySnapshot snapshot =
+      TestAggregateQuerySnapshot(aggregate_query, COUNT);
+
+  AggregateQuerySnapshot snapshot_copy_dest = snapshot;
+
+  EXPECT_EQ(snapshot.count(), COUNT);
+  EXPECT_EQ(snapshot.query(), aggregate_query);
+  EXPECT_TRUE(snapshot.is_valid());
+
+  EXPECT_EQ(snapshot_copy_dest.count(), COUNT);
+  EXPECT_EQ(snapshot_copy_dest.query(), aggregate_query);
+  EXPECT_TRUE(snapshot_copy_dest.is_valid());
+}
+
+TEST_F(AggregateQuerySnapshotTest, MoveConstructor) {
+  const Query& query = TestFirestore()->Collection("foo").Limit(10);
+  const AggregateQuery& aggregate_query = query.Count();
+
+  static const int COUNT = 8;
+  AggregateQuerySnapshot snapshot =
+      TestAggregateQuerySnapshot(aggregate_query, COUNT);
+
+  AggregateQuerySnapshot moved_snapshot_dest(std::move(snapshot));
+
+  EXPECT_EQ(snapshot.count(), 0);
+  EXPECT_EQ(snapshot.query(), AggregateQuery());
+  EXPECT_FALSE(snapshot.is_valid());
+
+  EXPECT_EQ(moved_snapshot_dest.count(), COUNT);
+  EXPECT_EQ(moved_snapshot_dest.query(), aggregate_query);
+  EXPECT_TRUE(moved_snapshot_dest.is_valid());
+}
+
+TEST_F(AggregateQuerySnapshotTest, MoveAssignmentOperator) {
+  const Query& query = TestFirestore()->Collection("foo").Limit(10);
+  const AggregateQuery& aggregate_query = query.Count();
+
+  static const int COUNT = 3;
+  AggregateQuerySnapshot snapshot =
+      TestAggregateQuerySnapshot(aggregate_query, COUNT);
+
+  AggregateQuerySnapshot snapshot_move_dest = std::move(snapshot);
+
+  EXPECT_EQ(snapshot.count(), 0);
+  EXPECT_EQ(snapshot.query(), AggregateQuery());
+  EXPECT_FALSE(snapshot.is_valid());
+
+  EXPECT_EQ(snapshot_move_dest.count(), COUNT);
+  EXPECT_EQ(snapshot_move_dest.query(), aggregate_query);
+  EXPECT_TRUE(snapshot_move_dest.is_valid());
+}
+
+TEST_F(AggregateQuerySnapshotTest, Equality1) {
```

Comment:
Please choose more descriptive test names than "Equality1", "Equality2", etc. The test names should describe the scenarios being testing. Personally, I use test naming convention "MethodNameShouldXXX". For example, "EqualsShouldBeTrueForIdenticalQueries"